severity.go

```diff
@@ -0,0 +1,51 @@
+package log
+
+import (
+	"fmt"
+	"strings"
+	"sync/atomic"
+)
+
+type Severity int32
+
+const (
+	SeverityInfo Severity = iota
+	SeverityWarning
+	SeverityError
+)
+
+var severityName = map[Severity]string{
+	SeverityInfo:    "INFO",
+	SeverityWarning: "WARN",
+	SeverityError:   "ERROR",
+}
+
+func (s *Severity) Get() Severity {
+	return Severity(atomic.LoadInt32((*int32)(s)))
+}
+
+func (s *Severity) Set(val Severity) {
+	atomic.StoreInt32((*int32)(s), int32(val))
+}
+
+func (s *Severity) Gte(val Severity) bool {
+	return s.Get() >= val
+}
+
+func (s Severity) String() string {
+	n, ok := severityName[s]
+	if !ok {
+		return "UNKNOWN SEVERITY"
+	}
+	return n
+}
+
+func SeverityFromString(s string) (Severity, error) {
```

Comment:
`Severity` and `SeverityFromString` do not need to be exported anymore. The latter never really did, and the former only needed to be exported for the `SetSeverity` function.